alchemist/alchemist-fxui/src/main/java/it/unibo/alchemist/boundary/gui/view/SingleRunApp.java

```diff
@@ -0,0 +1,459 @@
+import it.unibo.alchemist.model.interfaces.Position2D;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.event.ActionEvent;
+import javafx.scene.Scene;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static it.unibo.alchemist.boundary.gui.controller.ButtonsBarController.BUTTONS_BAR_LAYOUT;
+
+/**
+ * The class models a non-reusable GUI for simulation display.
+ *
+ * @param <T> the {@link it.unibo.alchemist.model.interfaces.Concentration} type
+ * @param <P> the position type
+ */
+@SuppressFBWarnings(value = "DM_EXIT", justification = "The program needs to exit when the user clicks on the exit button")
+public class SingleRunApp<T, P extends Position2D<P>> extends Application {
+    /**
+     * Main layout without nested layouts. Must inject eventual other nested layouts.
+     */
+    public static final String ROOT_LAYOUT = "RootLayout";
+    /**
+     * Effect pass param name.
+     */
+    public static final String USE_EFFECT_GROUPS_FROM_FILE = "use-effect-groups-from-file";
+    /**
+     * Default parameter start string.
+     */
+    public static final String PARAMETER_NAME_START = "--";
+    /**
+     * Default parameter end string.
+     */
+    public static final String PARAMETER_NAME_END = "=";
+    /**
+     * Default logger for the class.
+     */
+    private static final Logger L = LoggerFactory.getLogger(SingleRunApp.class);
+
+    private final Map<String, String> namedParams = new HashMap<>();
+    private final List<String> unnamedParams = new ArrayList<>();
+    private ObservableList<EffectGroup<P>> effectGroups = FXCollections.observableArrayList();
+    private boolean initialized;
+    @Nullable
+    private Simulation<T, P> simulation;
+    @Nullable
+    private FXOutputMonitor<T, P> displayMonitor;
+    private PlayPauseMonitor<T, P> playPauseMonitor;
+    private FXTimeMonitor<T, P> timeMonitor;
+    private FXStepMonitor<T, P> stepMonitor;
+
+    /**
+     * Method that launches the application.
+     * <br>
+     * For testing purpose only
+     *
+     * @param args {@link Parameters parameters} for the application
+     */
+    public static void main(final String[] args) {
```

Comment:
I do not understand why there is a main here. Launches should be controlled by `Alchemist`